File: 11-container-with-most-water/11-container-with-most-water.rb
# @param {Integer[]} height
# @return {Integer}
def max_area(height)
    max_vol = 0
    l = 0
    r = height.size - 1
    while l < r do 
        left = height[l]
        right = height[r]
        cur_vol = [left, right].min * (r - l)
        max_vol = [max_vol, cur_vol].max
        if left > right
            r -= 1
        else
            l += 1
        end
    end
    
    max_vol
end

File: medium/11-container-with-most-water/11-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
const maxArea = function(height) {
    let maxVol = 0,
        l = 0,
        r = height.length - 1;
    while (l < r) {
        const curVol = Math.min(height[l], height[r]) * (r - l);
        maxVol = Math.max(maxVol, curVol);
        height[l] < height[r] ? l++ : r--;
    }
    
    return maxVol;
}